action.yml

```diff
@@ -53,20 +53,47 @@ inputs:
 runs:
   using: 'composite'
   steps:
+    - name: Check if review is requested
+      shell: python
+      env:
+        PR_TITLE: ${{ github.event.pull_request.title }}
+        PR_STATE: ${{ github.event.pull_request.state }}
+        MERGE_PATTERN: \b(?:merged|closed)\b
+        PATTERN: \\b((?:no|skip)-(?:review|otter|otterai)|otter-(?:no|bye|restricted))(?:,((?:no|skip)-(?:review|otter|otterai)|otter-(?:no|bye|restricted)))*\\b
+      run: |
+        import os
+        import re
+
+        pr_title = os.environ['PR_TITLE']
+        pattern = os.environ['PATTERN']
+        merge_pattern = os.environ['MERGE_PATTERN']
+        pr_state = os.environ['PR_STATE']
+
+        if re.search(pattern, pr_title, re.IGNORECASE) and not re.search(merge_pattern, pr_state, re.IGNORECASE):
+            print("🦦 No review requested, skipping code review")
+            with open(os.environ['GITHUB_OUTPUT'], 'a') as f:
+                f.write("code_review_requested=false\n")
+        else:
+            print("🔍 Code review requested")
+            with open(os.environ['GITHUB_OUTPUT'], 'a') as f:
+                f.write("code_review_requested=true\n")
     - name: Set up Python
       uses: actions/setup-python@v5
+      if: steps.check_if_review_requested.outputs.code_review_requested != 'true'
       with:
         python-version: '3.12'
         cache: 'pip'
     
     - name: Install dependencies
+      if: steps.check_if_review_requested.outputs.code_review_requested != 'true'
       shell: bash
       run: |
         python -m pip install --upgrade pip
         pip install -r ${{ github.action_path }}/requirements.txt
 
     - name: Run OtterAI Code Review
       shell: bash
+      if: steps.check_if_review_requested.outputs.code_review_requested != 'true'
```

Comment:
🔍 The conditional check for running the code review is well-placed. Ensure that the outputs are correctly set in previous steps.